packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -84,55 +98,61 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
       scale = 1 / scale;
     }
 
-    return Scaffold(
-      body: ScannerOverlay(
-        restartCamera: _resumeImageStream,
-        stopCamera: _stopImageStream,
-        model: _model,
-        scannerWidget: Transform.scale(
-          scale: scale,
-          child: Center(
-            child: CameraPreview(
-              _controller!,
-            ),
-          ),
+    return Transform.scale(
+      scale: scale,
+      child: Center(
+        key: ValueKey<bool>(stoppingCamera),
+        child: CameraPreview(
+          _controller!,
         ),
       ),
     );
   }
 
-  Future<void> _startLiveFeed() async {
+  Future<void> startLiveFeed() async {
+    stoppingCamera = false;
     final CameraDescription camera = cameras[_cameraIndex];
-    _controller = CameraController(
+
+    final CameraController cameraController = CameraController(
       camera,
       ResolutionPreset.high,
       enableAudio: false,
     );
-    _controller!.setFocusMode(FocusMode.auto);
-    _controller!.lockCaptureOrientation(DeviceOrientation.portraitUp);
+    cameraController.setFocusMode(FocusMode.auto);
+    cameraController.lockCaptureOrientation(DeviceOrientation.portraitUp);
 
-    _controller!.initialize().then((_) {
-      if (!mounted) {
-        return;
+    _controller = cameraController;
+
+    // If the controller is updated then update the UI.
+    cameraController.addListener(() {
+      if (mounted) {
+        setState(() {});
+      }
+      if (cameraController.value.hasError) {
+        debugPrint(cameraController.value.errorDescription);
       }
-      _controller!.startImageStream(_processCameraImage);
-      imageStreamActive = true;
-      setState(() {});
     });
-  }
 
-  void _resumeImageStream() {
-    if (_controller != null && !imageStreamActive) {
-      _controller!.startImageStream(_processCameraImage);
-      imageStreamActive = true;
+    try {
+      await cameraController.initialize();
+      _controller?.startImageStream(_processCameraImage);
+    } on CameraException catch (e) {
+      if (kDebugMode) {
+        debugPrint(e.toString());
+      }
```

Comment:
We probably need something else.
If you don't know what to do now, please add a `TODO`.